File: analytics/src/main/java/io/appmetrica/analytics/impl/utils/validation/ValidationResult.java
package io.appmetrica.analytics.impl.utils.validation;

import androidx.annotation.NonNull;

public final class ValidationResult {

    private final Class<? extends Validator> mValidatorClass;
    private final boolean mValid;
    private final String mDescription;

    private ValidationResult(@NonNull Validator<?> validator,
                             boolean valid,
                             @NonNull String description) {
        mValidatorClass = validator.getClass();
        mValid = valid;
        mDescription = description;
    }

    public final boolean isValid() {
        return mValid;
    }

    @NonNull
    public final String getDescription() {
        return mDescription;
    }

    @NonNull
    public Class<? extends Validator> getValidatorClass() {
        return mValidatorClass;
    }

    public static final ValidationResult successful(@NonNull Validator<?> validator) {
        return new ValidationResult(validator, true, "");
    }

    public static final ValidationResult failed(@NonNull Validator<?> validator, @NonNull String description) {
        return new ValidationResult(validator, false, description);
    }
}

File: core-utils/src/main/java/io/appmetrica/analytics/coreutils/internal/validation/ValidationResult.kt
package io.appmetrica.analytics.coreutils.internal.validation

class ValidationResult private constructor(
    validator: Validator<*>,
    val isValid: Boolean,
    val description: String
) {
    val validatorClass = validator.javaClass

    companion object {

        @JvmStatic
        fun successful(validator: Validator<*>): ValidationResult {
            return ValidationResult(validator, true, "")
        }

        @JvmStatic
        fun failed(validator: Validator<*>, description: String): ValidationResult {
            return ValidationResult(validator, false, description)
        }
    }
}
